terraform/eval_refresh.go

```diff
@@ -18,56 +18,54 @@ import (
 
 // EvalRefresh is an EvalNode implementation that does a refresh for
 // a resource.
-type EvalRefresh struct {
+type EvalRefreshRequest struct {
 	Addr           addrs.ResourceInstance
 	ProviderAddr   addrs.AbsProviderConfig
 	Provider       *providers.Interface
 	ProviderMetas  map[addrs.Provider]*configs.ProviderMeta
-	ProviderSchema **ProviderSchema
-	State          **states.ResourceInstanceObject
-	Output         **states.ResourceInstanceObject
+	ProviderSchema *ProviderSchema
+	State          *states.ResourceInstanceObject
 }
 
 // TODO: test
-func (n *EvalRefresh) Eval(ctx EvalContext) tfdiags.Diagnostics {
-	state := *n.State
-	absAddr := n.Addr.Absolute(ctx.Path())
-
+func Refresh(req *EvalRefreshRequest, ctx EvalContext) (*states.ResourceInstanceObject, tfdiags.Diagnostics) {
```

Comment:
I think all the other "actions" are now methods on their respective types. Would it make sense to take this and move it into a method on a resource type?

Reply:
I like it, I'll give that a look!